lib/AST/ASTMangler.cpp

```diff
@@ -548,6 +548,9 @@ void ASTMangler::appendDeclName(const ValueDecl *decl) {
     case DeclBaseName::Kind::Subscript:
       appendIdentifier("subscript");
       break;
+    case DeclBaseName::Kind::Destructor:
+      // FIXME: Is it true that this is unreachable?
```

Comment:
I'm 90% sure this is true but try defining a local type inside a destructor to make sure.